lib/fizzy/execute.cpp

```diff
@@ -176,6 +176,16 @@ Instance instantiate(const Module& module)
     // NOTE: fill it with zeroes
     bytes memory(memory_min * page_size, 0);
 
+    // Fill out memory based on data segments
+    for (auto const& data : module.datasec)
+    {
+        // FIXME: assert in parser
+        assert(data.memidx == 0);
```

Comment:
I would do it this way for now, but when we have spec tests and clean up the code, we can move a lot of these "wasm version 1" assumptions into the parser.

Reply:
@chfast @gumb0 I am changing it to this, unless you guys object.

Reply:
"to this" mean moving it to parser?

Reply:
Yeah.